.github/CONTRIBUTING.md

```diff
@@ -118,45 +119,83 @@ This project uses [Prettier](http://prettier.io/) for code formatting. Consider
 
 ## Commiting
 
-- Never use the `-m <msg>` / `--message=<msg>` flag to `git commit`;
+A commit message can consists of a **header**, **body** and **footer**. The header is the only mandatory part and consists of a type and a subject. The body is used to fully describe the change. The footer is the place to reference any issues or pull requests related to the commit. That said, we end with a template like this:
+
+```
+<type>: <subject>
+
+[optional body]
+
+[optional footer]
+```
+
+To ensure that a commit is valid, easy to read, and changelog-ready, we have a hook that lints the commit message before allowing a commit to pass. This linter verifies the following:
+
+- The header (first line) is the only mandatory part of the commit message.
+- The header should contains:
+  - A type:
+    - Must be lowercase;
+    - Must be one of:
+      - **chore**: A change that neither fix a bug or adds a feature;
+      - **ci**: A CI change;
+      - **docs**: A documentation change or fix;
+      - **feat**: A new feature;
+      - **fix**: A bug fix;
+      - **test**: A test-related change.
+  - A subject:
+    - Must be capitalized;
+    - Must be limited to 50 characters or less;
+    - Must omit any trailing punctuation;
+- Header, body and footer must be separated by a blank line;
+- The following lines should have 72 characters or less;
```

Comment:
You have a section `Header should contains`, i think that repeating the same name pattern for `body` and `footer` is a good idea, i had to scroll down the docs to know that referencing goes on footer because it's not clear here.

Reply:
Eh, since they both are optional and only check for leading blank line and characters limit, I thought it would be pretty redundant and borderline useless.

It would be something like this:
---
Remove:
- Header, body and footer must be separated by a blank line;
- The following lines should have 72 characters or less;
- Reference to issues and pull requests must be made in the last line.

Add
- The body:
  - Must have a leading blank line;
  - Each line must be limited to 72 characters or less.
- The footer:
  - Must have a leading blank line;
  - Each line must be limited to 72 characters or less;
  - If needed, reference to issues and pull requests must be made here in the last line.

Reply:
I think it's better this way with separated topics for header, body and footer.